crates/host/src/wasmbus/mod.rs

```diff
@@ -1111,13 +1228,51 @@ impl Lattice {
             })
             .collect()
             .await;
+        let providers: Vec<_> = self
+            .providers
+            .read()
+            .await
+            .iter()
+            .flat_map(
+                |(
+                    public_key,
+                    Provider {
+                        claims, instances, ..
+                    },
+                )| {
+                    instances.iter().filter_map(
+                        move |(link_name, ProviderInstance { child, id })| {
+                            let pid = child.id()?;
+                            let instance_id = Uuid::from_u128((*id).into());
+                            if let Some(jwt::CapabilityProvider { capid, .. }) = &claims.metadata {
+                                Some(json!({
+                                    "public_key": public_key,
+                                    "link_name": link_name,
+                                    "contract_id": capid,
```

Comment:
Same comment about not duplicating the JSON as I made before. I prefer to not have to visually diff the structures